src/catchup/CatchupWork.cpp

```diff
@@ -463,6 +463,14 @@ CatchupWork::runCatchupStep()
             return res;
         }
     }
+    {
+        LedgerTxn ltx(mApp.getLedgerTxnRoot());
+        if (protocolVersionStartsFrom(ltx.loadHeader().current().ledgerVersion,
+                                      SOROBAN_PROTOCOL_VERSION))
+        {
+            mApp.getLedgerManager().updateNetworkConfig(ltx);
```

Comment:
All the changes here are made to make tests pass. IIUC here the reason is that we just load the buckets, but haven't closed any ledgers yet. We need the config *before* we close the first ledger though, in order to be able to apply the buffered tx set.

Reply:
I think letting the application get this far before loading the config is error-prone. Catchup work isn't any different in the sense that it still requires application startup, so my question is how come we don't have the config loaded then?

Reply:
I'm not sure I understand; when you start catchup you don't necessarily have a config because you're on, say, p19. Even if you're on p20, you still need to update your config, because it won't necessarily match what you have on startup.

The config has to be updated any time ledger state changes; it's not just about loading it.

Reply:
The config should only be updated if either we applied a new ledger that changed the config, or if we starting the node and need to load it to memory from the DB, right? In v19, there should be no config, as it will get initialized when we apply the first v20 ledger. In v20, when we start the node we should update the config based on persisted LCL. Later, when we apply a new v20 ledger that potentially changes the config, it will get refreshed in `closeLedger`. Maybe we actually have a bug that uses the config when it shouldn't, which causes tests to fail?